src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/ReviewTimeExtensionsHandler.java

```diff
@@ -79,8 +148,32 @@ private String getTimeExtensionDecisionReason(AsylumCase asylumCase) {
         return read.orElseThrow(() -> new IllegalArgumentException("Cannot handle " + Event.REVIEW_TIME_EXTENSION + " without a decision reason"));
     }
 
-    private Stream<IdValue<TimeExtension>> getTimeExtensions(AsylumCase asylumCase) {
+    private String getTimeExtensionDueDate(AsylumCase asylumCase) {
+        Optional<String> read = asylumCase.read(REVIEW_TIME_EXTENSION_DUE_DATE);
+        return read.orElseThrow(() -> new IllegalArgumentException("Cannot handle " + Event.REVIEW_TIME_EXTENSION + " without a decision reason"));
+    }
+
+    private List<IdValue<PreviousDates>> appendPreviousDates(List<IdValue<PreviousDates>> previousDates, String dateDue, String dateSent) {
+
+        if (CollectionUtils.isEmpty(previousDates)) {
+
+            return newArrayList(new IdValue<>("1", new PreviousDates(dateDue, dateSent)));
+        } else {
+
+            int newEntryId = previousDates.size() + 1;
+            List<IdValue<PreviousDates>> newEntry = Collections.singletonList(
+                new IdValue<>(String.valueOf(newEntryId),
+                    new PreviousDates(dateDue, dateSent)));
+
+            Stream<IdValue<PreviousDates>> combinedStream = Stream.of(newEntry, previousDates)
+                .flatMap(Collection::stream);
+
+            return combinedStream.collect(Collectors.toList());
+        }
+    }
```

Comment:
If this is duplicated from the update direction due date stuff it needs to be pulled out to a common class as it needs to be done the same way as when we update the due date normally.